src/thor/costmatrix.cc

```diff
@@ -16,6 +16,7 @@ namespace {
 
 constexpr uint32_t kMaxMatrixIterations = 2000000;
 constexpr uint32_t kMaxThreshold = std::numeric_limits<int>::max();
+constexpr uint32_t kMaxLocationCache = 20;
```

Comment:
I'm actually thinking about making this configurable. It goes a loooong way to speed up the computations and for people who are using Valhalla for matrix computations mainly, it makes sense to reserve as much memory as possible.

Reply:
nit: cache is a poor name here, it should be something about "reservation". cache implies that it holds data that we want to use again, when in fact it simply holds an allocation that we want to re-use with new data